zproject/backends.py

```diff
@@ -826,8 +853,9 @@ def get_or_build_user(self, username: str, ldap_user: _LDAPUser) -> Tuple[UserPr
             opts["default_stream_groups"] = []
 
         user_profile = do_create_user(
-            username, None, self._realm, full_name, acting_user=None, **opts
+            username, None, self._realm, full_name, acting_user=None, ldap_auth_id=dn, **opts
```

Comment:
I'm not sure if we should be adding another argument to do_create_user, when we can leave the functions as it is, and set the ldap dn on the UserProfile right after the call:
```
user_profile = do_create_user(...)
do_set_ldap_unique_identifier(user, ...)
...
```

Reply:
Yeah, I think that's probably the right model.